src/configuration.ts

```diff
@@ -166,22 +166,14 @@ export function loadConfigurationFromPath(configFilePath: string): IConfiguratio
  * @var relativeFilePath Relative path or package name (tslint-config-X) or package short name (X)
  */
 function resolveConfigurationPath(relativeFilePath: string, relativeTo?: string) {
-    let resolvedPath: string;
-    if (pathIsAbsolute(relativeFilePath)) {
-        resolvedPath = relativeFilePath;
-    } else if (relativeFilePath.indexOf(".") === 0) {
-        resolvedPath = getRelativePath(relativeFilePath, relativeTo);
-    } else {
-        try {
-            resolvedPath = require.resolve(relativeFilePath);
-        } catch (err) {
-            throw new Error(`Invalid "extends" configuration value - could not require "${relativeFilePath}". ` +
-                "Review the Node lookup algorithm (https://nodejs.org/api/modules.html#modules_all_together) " +
-                "for the approximate method TSLint uses to find the referenced configuration file.");
-        }
+    const basedir = relativeTo || process.cwd();
```

Comment:
That would mean it's not the same as the [`require.resolve() algorithm](https://nodejs.org/dist/latest-v4.x/docs/api/modules.html#modules_all_together). Before this pull request `Y` (in the algorithm) was always the location of the tslint package, now it's the location of the parent `tslint.json` file (or the tslint package the first time).

It's similar to how you'd expect `node -e "require('foo')"` to fail if `foo` was installed globally but not locally.

However, it's arguably a breaking change, so I'm happy to add a fallback which resolves with `require.resolve` to avoid issues for other users.